rfcs/030-pipeline_merging/README.md

```diff
@@ -0,0 +1,124 @@
+elastic search greater or equal than versioning, the unlinked version would have been 
+ingested (provided no message out of order issues)
+
+
+### Problems
+
+There are a number of issues with the current approach:
+- We need to multiply `transformerVersion` by 1000 to make it take precedence 
+over `numberOfSources` because `numberOfSources` can be quite high (650 is the recorded max).
+ If we don't do that we run into data consistency problems in the index. 
+ As we add more sources we will find ourselves incrementing this number regularly
+- The unliking case only works if the update that causes the unlink is on the target work,
+ therefore incrementing `transformerVersion`. If, as in the example above, the update is on a
+  work that gets redirected to the target, it won't be reflected in the API
+- The ingestor is responsible for figuring out the version based on information passed on
+ by the merger. This causes coupling between the merger and the ingestor. The ingestor is 
+ currently aware of merging happening at some point, which it shouldn't be. It also makes the 
+ versioning logic very hard to follow and modify
+
+### Discarded ideas
+
+Jamie and I came up with multiple ideas to tweak the current behaviour, each one with some problems:
+- Go back to the `merged` flag: it has issues with multiple sources and it has issues 
+with images versioning because image versions are derived from works
+- Add unlink in the version function like 
+`(transformerVersion * 10000) + (unlinked?1:0)*1000 + numberOfSources`: 
+bleurgh and also when something is ingested as unlinked it's impossible to override 
+it unless the `transformerVersion` changes
+- Use something like the sum of all `transformerVersion`s of all works sent by the matcher: 
+this would succeed in unlinking, but again would fail to add a work to a group 
+that was the result of a previous unlink operation
+
+
+### Solution proposed
+
+The solution suggested by Nic is that we pass through in the work a field with the timestamp 
+of the last update.
+The merger should then use the maximum of the timestamps of all works sent by the matcher 
+as the version for each of the works.
+The ingestor should use greater than versioning instead of greater or equal.
+ 
+The updated timestamp needs to be recorded in the adapter in different ways depending on the source:
+- Sierra: the sierra API already provides a timestamp with down to the second granularity 
+for each update and the Sierra adapter already stores it in the Sierra VHS. The sierra transfomer can 
+read this value and copy it into the work. 
+- Miro: Miro data doesn't change so can just set it to zero in the transformer
+- Calm: Calm data has a `retrievedAt` field in the Calm VHS. It comes from the HTTP header 
+of the response from the Calm API and it just represents when the response was sent from the server. 
+Because of this a lot of Calm records have the same timestamp of when the first harvest was done. 
```

Comment:
Looking at the code it looks to me like the `retrievedAt` field is initialised from the HTTP request for the individual Calm record, which would mean they would have different timestamps?

Reply:
If I undesrand the code, it comes from the response, not the request. But yeah, it's not exactly the same so probably no reason to worry and will update

Reply:
Yep it's the response :+1: